### ISSUE TITLE
More Death Types for Air Vehicles

### ISSUE THREAD

**Issue by MavroGada2** [TheSuperHackers/GeneralsGamePatch#1650]
Different dying behaviors for air vehicles can be good like other units which had.

### Comanche with normal death type(Original):

https://user-images.githubusercontent.com/103841664/217869582-4f5d2412-1c24-46b8-ba5a-227602948829.mp4

### Comanche with explosive death type:

https://user-images.githubusercontent.com/103841664/217871901-b338b7b7-2003-4eb2-8abf-3e3e255c611d.mp4

### Raptor with normal death type(Original):

https://user-images.githubusercontent.com/103841664/217877315-ac6d7155-0d17-411a-b629-406117f4bd7e.mp4

### Raptor with explosive death type:

https://user-images.githubusercontent.com/103841664/217877444-681f6597-26b9-45de-861d-8fa8cca97c5b.mp4

**Comment by xezon** [comment]
The exploding Raptor looks great. Not so sure about the exploding Comanche.

**Comment by MavroGada2** [comment]
I have better plans for comanche. (devs didn't finish propeller model but I putted it for experiment )



https://user-images.githubusercontent.com/103841664/217913565-1e766548-e702-4777-8c92-7e1d1ea213e9.mp4


Hollywood style escape:


https://user-images.githubusercontent.com/103841664/217913787-950475f2-7b86-4e28-af0e-30aaf951e2c3.mp4

**Comment by MavroGada2** [comment]
Use of object with slowdeath instead of debris:


https://user-images.githubusercontent.com/103841664/218045271-d28675bc-086c-4388-8ef9-554a806ed0e0.mp4

**Comment by xezon** [comment]
Looks nice. Can you post the codes for Raptor, Comanche, (Chinook, Helix)?

**Comment by MavroGada2** [comment]
I didn't make for other helis yet. And changed raptor debris to object for hit the ground effects.

[AirBomb.txt](https://github.com/TheSuperHackers/GeneralsGamePatch/files/10707229/AirBomb.txt)

**Comment by MavroGada2** [comment]
Because of heavy swing animation of Helix, the transition from life to death doesn't flow like comanche.

I wanted to give the feeling that something really big is dying.

Now i noticed i didn't add shadow to dying objects.


https://user-images.githubusercontent.com/103841664/218118037-170ed262-09db-436d-8767-ccab7a2ad547.mp4

**Comment by MavroGada2** [comment]
Chinook is losing front propeller and fall accordingly. Shadows and objects added to all. 




https://user-images.githubusercontent.com/103841664/218143155-e26e0734-e4da-46a2-9cb2-43ebd309acdb.mp4


File:

[AirBomb.txt](https://github.com/TheSuperHackers/GeneralsGamePatch/files/10709351/AirBomb.txt)

**Comment by xezon** [comment]
The transition between flying and death does not look good. The original transition looks much better.

**Comment by MavroGada2** [comment]
Yes but  we can add another helislowdeath or jetslowdeath modules which working correctly with locomotor. I tried something before but didn't get good result. I will look at.

**Comment by MavroGada2** [comment]
Jetslowdeath;


https://user-images.githubusercontent.com/103841664/218254698-1fc6bb6a-eddc-44d9-a64d-3114f10ceeb3.mp4

**Comment by xezon** [comment]
That looks much better indeed. There is still a small hickup on crash. Not sure if this is as per original behaviour.

**Comment by MavroGada2** [comment]
Because of roll effect from jetslowdeath, default ocl is not spawn at the last moment of the helicopter. I wrote new coordination for ocl and watched frame by frame, new ocl is very close  to helix's last position but other groundhit effects not in the center anymore because of new coordinations.

https://user-images.githubusercontent.com/103841664/218258175-abde98f0-d2df-4bb9-8f08-63f6cdceee67.mp4


### Helislowdeath:

Not good but a option

https://user-images.githubusercontent.com/103841664/218258253-10425b7f-995a-482e-96d9-3696c5adb896.mp4
